packages/flutterfire_cli/lib/src/commands/install.dart

```diff
@@ -0,0 +1,326 @@
+      'Select the Firebase plugins you would like to install',
+      choices,
+      defaultSelection: defaultSelection,
+    );
+    for (final index in selectedChoices) {
+      selectedPlugins.add(FlutterFirePlugins.values[index]);
+    }
+    return selectedPlugins;
+  }
+
+  Future<Map<String, String>> _getPluginVersionsFromJSON(
+    String bomVersion,
+  ) async {
+    // TODO: change to master
+    const bomPath =
+        'https://raw.githubusercontent.com/firebase/flutterfire/chore/proposal/scripts/versions.json';
+
+    final http = HttpClient();
+    final request = await http.getUrl(Uri.parse(bomPath));
+    final response = await request.close(); // sends the request
+    final jsonString = await response.transform(utf8.decoder).join();
+    final json = jsonDecode(jsonString) as Map<String, dynamic>;
+
+    if (bomVersion == 'latest' ||
+        bomVersion == 'master' ||
+        bomVersion == 'main' ||
+        bomVersion.contains('git')) {
+      // The latest version is the first key in the JSON
+      return ((json[json.keys.first] as Map)['packages'] as Map)
+          .cast<String, String>();
+    }
+
+    if (!json.containsKey(bomVersion)) {
+      throw Exception(
+        'BoM version $bomVersion not found. Check the available versions at https://github.com/firebase/flutterfire/blob/chore/proposal/VERSIONS.md',
+      );
+    }
+
+    return ((json[bomVersion] as Map)['packages'] as Map)
+        .cast<String, String>();
+  }
+
+  @override
+  Future<void> run() async {
+    // Get the BoM version number from the arguments
+    if (argResults?.arguments.length != 1) {
+      stderr.writeln(
+        'Usage for install command: flutterfire install <version>',
+      );
+      return;
+    }
+    final bomVersion = argResults!.arguments[0];
+
+    try {
+      commandRequiresFlutterApp();
+
+      final pluginVersions = await _getPluginVersionsFromJSON(bomVersion);
+
+      final selectedPlugins = await _selectPlugins(pluginVersions);
+
+      final pubSpec = flutterApp!.package.pubSpec;
+      final listOfAlreadyInstalledPlugins = pubSpec.dependencies.keys
+          .where(
+            (element) =>
+                FlutterFirePlugins.allPluginsPublicNames.contains(element),
+          )
+          .toList();
+
+      final pluginsToDelete = listOfAlreadyInstalledPlugins
+          .where(
+            (element) =>
+                !selectedPlugins.any((plugin) => plugin.name == element),
+          )
+          .toList();
+
+      if (pluginsToDelete.isNotEmpty) {
+        stdout.writeln('The following plugins will be removed:');
+        for (final plugin in pluginsToDelete) {
+          stdout.writeln(' - $plugin');
+        }
+        final removingSpinner = spinner(
+          (done) {
+            if (!done) {
+              return 'Removing plugins ... ';
+            }
+            return 'Plugins removed.';
+          },
+        );
+
+        await Process.run(
+          'flutter',
```

Comment:
This should be compatible with Windows, we have an open issue for the `flutterfire update` command because this doesn't work on windows, it should possible be `flutter.bat`: https://github.com/invertase/flutterfire_cli/issues/252#issuecomment-1958830688

also should be updated on other `flutter` commands.

Reply:
Let me try on my windows machine :)

Reply:
I just tried and it wasn't working with `flutter pub add` but working fine with `dart pub add`. Since they have the same features I've moved to `dart` :)